### PR TITLE
Add .npmignore for smaller builds

### PR DESCRIPTION
**Pull request by ruimarinho**

### CODE REVIEW

File: .npmignore [hunk 1]
```diff
@@ -0,0 +1,3 @@
+.gitignore
```
Comment (kurayama) [inline .npmignore]: Not needed

Comment (ruimarinho) [inline .npmignore]: Source?

> By default, the following paths and files are ignored, so there's no need to add them to .npmignore explicitly:

> .*.swp
> ._*
> .DS_Store
> .git
> .hg
> .npmrc
> .lock-wscript
> .svn
> .wafpickle-*
> config.gypi
> CVS
> npm-debug.log

From: https://docs.npmjs.com/misc/developers

Comment (kurayama) [inline .npmignore]: [Sequelize](https://github.com/sequelize/sequelize/blob/master/.npmignore) for example does not include that line and the package does not contain the `.gitignore` file.

You can test it by: 
 - removing the line
 - `npm pack`
 - `tar -xzf eslint-config-seegno-7.0.0.tgz`
 - `ls -la package`

Comment (ruimarinho) [inline .npmignore]: Which npm version are they using?

Comment (kurayama) [inline .npmignore]: Dunno, I tested locally using node 6.7 and 7 nightly.

Comment (ruimarinho) [inline .npmignore]: This is the culprit and on a first look I don't think it was intended to have this effect https://github.com/npm/fstream-npm/blob/master/fstream-npm.js#L331-L336

**Pull request merged**